python/lsst/afw/image/exposureInfo.cc

```diff
@@ -84,126 +85,129 @@ void declareGenericMethodsMerged(PyExposureInfo &cls) {
-    cls.def("hasPsf", &ExposureInfo::hasPsf);
-    cls.def("getPsf", &ExposureInfo::getPsf);
-    cls.def("setPsf",
-            [](ExposureInfo &self, py::object psf) {
-                if (psf.is(py::none())) {
-                    self.setPsf(nullptr);
-                } else {
-                    self.setPsf(py::cast<std::shared_ptr<afw::detection::Psf>>(psf));
-                }
-            },
-            "psf"_a);
-
-    cls.attr("KEY_VALID_POLYGON") = ExposureInfo::KEY_VALID_POLYGON.getId();
-    cls.def("hasValidPolygon", &ExposureInfo::hasValidPolygon);
-    cls.def("getValidPolygon", &ExposureInfo::getValidPolygon);
-    cls.def("setValidPolygon",
-            [](ExposureInfo &self, py::object polygon) {
-                if (polygon.is(py::none())) {
-                    self.setValidPolygon(nullptr);
-                } else {
-                    self.setValidPolygon(py::cast<std::shared_ptr<afw::geom::polygon::Polygon>>(polygon));
-                }
-            },
-            "polygon"_a);
-
-    cls.attr("KEY_AP_CORR_MAP") = ExposureInfo::KEY_AP_CORR_MAP.getId();
-    cls.def("hasApCorrMap", &ExposureInfo::hasApCorrMap);
-    cls.def("getApCorrMap", (std::shared_ptr<ApCorrMap>(ExposureInfo::*)()) & ExposureInfo::getApCorrMap);
-    cls.def("setApCorrMap", &ExposureInfo::setApCorrMap, "apCorrMap"_a);
-    cls.def("initApCorrMap", &ExposureInfo::initApCorrMap);
-
-    cls.attr("KEY_COADD_INPUTS") = ExposureInfo::KEY_COADD_INPUTS.getId();
-    cls.def("hasCoaddInputs", &ExposureInfo::hasCoaddInputs);
-    cls.def("getCoaddInputs", &ExposureInfo::getCoaddInputs);
-    cls.def("setCoaddInputs", &ExposureInfo::setCoaddInputs, "coaddInputs"_a);
-
-    cls.def("hasVisitInfo", &ExposureInfo::hasVisitInfo);
-    cls.def("getVisitInfo", &ExposureInfo::getVisitInfo);
-    cls.def("setVisitInfo", &ExposureInfo::setVisitInfo, "visitInfo"_a);
-
-    cls.attr("KEY_TRANSMISSION_CURVE") = ExposureInfo::KEY_TRANSMISSION_CURVE.getId();
-    cls.def("hasTransmissionCurve", &ExposureInfo::hasTransmissionCurve);
-    cls.def("getTransmissionCurve", &ExposureInfo::getTransmissionCurve);
-    cls.def("setTransmissionCurve", &ExposureInfo::setTransmissionCurve, "transmissionCurve"_a);
+    lsst::utils::python::WrapperCollection wrappers(mod, "lsst.afw.image.exposureInfo");
+    wrappers.addInheritanceDependency("lsst.daf.base");
+    wrappers.addInheritanceDependency("lsst.afw.geom");
+    wrappers.addInheritanceDependency("lsst.afw.cameraGeom.detector");
+    wrappers.addInheritanceDependency("lsst.afw.detection");  // For Psf
+    wrappers.addInheritanceDependency("lsst.afw.image.photoCalib");
+    wrappers.addInheritanceDependency("lsst.afw.image.apCorrMap");
+    wrappers.addInheritanceDependency("lsst.afw.image.coaddInputs");
+    wrappers.addInheritanceDependency("lsst.afw.image.filter");
+    wrappers.addInheritanceDependency("lsst.afw.image.filterLabel");
+    wrappers.addInheritanceDependency("lsst.afw.image.visitInfo");
```

Comment:
I hope all or at least most of these can actually be signature dependencies instead.